src/term/mod.rs

```diff
@@ -104,9 +153,8 @@ impl<'a> Drop for RenderableCellsIter<'a> {
     /// Resets temporary render state on the grid
     fn drop(&mut self) {
         if self.cursor_is_visible() {
-            if let Some(ref original) = self.cursor_original {
-                self.grid[self.cursor] = original.inner;
-            }
+            let cell = self.cursor_cell.as_mut();
+            cell.inner = self.grid[self.cursor];
```

Comment:
Shouldn't this be updating the cell in the grid instead of updating the cell that's getting dropped? Additionally, I think this should only happen when `cursor_cell` is `CursorCell::Replacement`.